build.sbt

```diff
@@ -63,3 +64,14 @@ lazy val scoverageSettings: Seq[Setting[_]] = Seq(
   coverageHighlighting := true,
   parallelExecution in Test := false
 )
+
+lazy val silencerSettings: Seq[Setting[_]] = {
+  val silencerVersion = "1.7.0"
+  Seq(
+    libraryDependencies ++= Seq(compilerPlugin("com.github.ghik" % "silencer-plugin" % silencerVersion cross CrossVersion.full)),
+    // silence all warnings on autogenerated files
+    scalacOptions += "-P:silencer:pathFilters=target/.*",
+    // Make sure you only exclude warnings for the project directories, i.e. make builds reproducible
+    scalacOptions += s"-P:silencer:sourceRoots=${baseDirectory.value.getCanonicalPath}"
+  )
+}
```

Comment:
Is this necessary?

Reply:
This is supressing (mostly invalid) warnings from twirl templates and auto-generated files. This seems to be a bug, with no intent to fix it: https://github.com/playframework/playframework/issues/7382